src/helper.h

```diff
@@ -212,6 +215,9 @@ class SessionBase: public CbdataParent
     /// whether the server is locked for exclusive use by a client
     virtual bool reserved() = 0;
 
+    /// our creator (parent) object
+    virtual Client &helper() = 0;
```

Comment:
I hope we can get away with this. If not, we would need to provide two getters, one const and one not.

```suggestion
    virtual Client &helper() const = 0;
```

We need a const getter so that future code can use helper() in constant SessionBase methods and similar const contexts...